be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -47,7 +52,63 @@ case object RollCallHandler extends MessageHandler {
+        def createAttendeeChannels(attendees: List[PublicKey], rpcMessage: JsonRpcRequest): GraphMessage = {
+          attendees match {
+            case Nil => Left(rpcMessage)
+            case head::tail => 
+              //a closeRollCall is always sent in the lao's main channel so we are allowed to do this
+              val socialChannel: String = rpcMessage.getParamsChannel + "social/" + head.toString
+              val ask: Future[GraphMessage] = (dbActor ? DbActor.CreateChannel(Channel(socialChannel), ObjectType.CHIRP)).map {
+                case DbActorWriteAck() => createAttendeeChannels(tail, rpcMessage)
+                //FIXME: Once the real PoP tokens are implemented, the same person will not have access to the same channel name twice in a row, so we can remove this line
+                //however, this is to prevent errors for now within a LAO for multiple Roll Calls and the same participants since the public key will probably not change
+                case DbActorNAck(code, description) if description == "Database cannot create an already existing channel (" + socialChannel + ")" => createAttendeeChannels(tail, rpcMessage)
+                case DbActorNAck(code, description) => Right(PipelineError(code, description, rpcMessage.id))
+                case _ => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, unknownAnswer, rpcMessage.id))
+              }
+              Await.result(ask, duration)
+          }
+        }
+
+        val laoChannel: Option[Array[Byte]] = rpcMessage.getParamsChannel.decodeSubChannel
+        laoChannel match {
+          case Some(channel) =>
+          case None => Right(PipelineError(
+            ErrorCodes.SERVER_ERROR.id,
+            s"There is an issue with the data of the LAO",
+            rpcMessage.id
+          ))
+        }
+
+        val askOldData = dbActor ? DbActor.ReadLaoData(rpcMessage.getParamsChannel)
+        
+        Await.result(askOldData, duration) match {
+          case DbActorReadLaoDataAck(Some(oldLaoData)) =>
+            val laoData: LaoData = LaoData(oldLaoData.owner, data.attendees)
+            val ask: Future[GraphMessage] = (dbActor ? DbActor.WriteLaoData(rpcMessage.getParamsChannel, message, laoData)).map {
+              case DbActorWriteAck() => createAttendeeChannels(data.attendees, rpcMessage)
+              case DbActorNAck(code, description) => Right(PipelineError(code, description, rpcMessage.id))
+              case _ => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, unknownAnswer, rpcMessage.id))
+            }
+            Await.result(ask, duration)
+          case _ => Right(PipelineError(
+            ErrorCodes.SERVER_ERROR.id,
+            s"There is an issue with the data of the LAO",
+            rpcMessage.id
+          ))
+
+        }
+      case _ => Right(PipelineError(
+        ErrorCodes.SERVER_ERROR.id,
+        s"Unable to handle lao message $rpcMessage. Not a Publish/Broadcast message",
+        rpcMessage.id
+      ))
+    }
```

Comment:
Feels good to see this come to life 😃 after some tears.

Reply:
:)